lib/active_model/serializer/concerns/associations.rb

```diff
@@ -83,15 +83,17 @@ def associate(reflection)
       #   +default_include_directive+ config value when not provided)
       # @return [Enumerator<Association>]
       #
-      def associations(include_directive = ActiveModelSerializers.default_include_directive)
+      def associations(include_directive = ActiveModelSerializers.default_include_directive, include_slice = nil)
+        include_slice ||= include_directive
         return unless object
 
         Enumerator.new do |y|
           self.class._reflections.values.each do |reflection|
             next if reflection.excluded?(self)
             key = reflection.options.fetch(:key, reflection.name)
             next unless include_directive.key?(key)
```

Comment:
This is one of the gross aspects of the current adapter but

* we get the [fields](https://github.com/rails-api/active_model_serializers/blob/master/lib/active_model_serializers/adapter/json_api.rb#L306), defaulted to '*'.

* they end up in [relationships_for](https://github.com/rails-api/active_model_serializers/blob/master/lib/active_model_serializers/adapter/json_api.rb#L435)

* that calls [#associations](https://github.com/rails-api/active_model_serializers/blob/master/lib/active_model_serializers/adapter/json_api.rb#L440) leading us to this code.

That's opposed to `include_slice`, which is always the slice of the *real* include directive passed to `render`.

In other words, the current code has two 'include directives'. One real one that is what you think it is, and a separate one that is derived from a relationship's `fields`